internal/service/api/requests/get_role.go

```diff
@@ -17,14 +18,14 @@ func NewGetRoleRequest(r *http.Request) (GetRoleRequest, error) {
 
 	err := urlval.Decode(r.URL.Query(), &request)
 	if err != nil {
-		return request, err
+		return request, errors.Wrap(err, "failed to decode url")
 	}
 
-	return request, validateGetRoleRequest(request)
+	return request, request.validate()
 }
 
-func validateGetRoleRequest(request GetRoleRequest) error {
+func (r *GetRoleRequest) validate() error {
 	return validation.Errors{
-		"accessLevel": validation.Validate(request.AccessLevel, validation.Required, validation.In(data.GetRoles()...)),
+		"accessLevel": validation.Validate(r.AccessLevel, validation.Required, validation.In(data.GetRoles()...)),
```

Comment:
Have you tested it? If i'm not mistaken, `.In` takes interface, so it could work not as expected, just to test in future

Reply:
It takes a slice of interfaces. Tested it - works just fine